gp-includes/formats/format-android.php

```diff
@@ -44,108 +100,485 @@ public function print_exported_file( $project, $locale, $translation_set, $entri
 				$mapping[ $array_name ] = array();
+			}
 
-			// Because Android doesn't fallback on the original locale
+			// Because Android doesn't fall back on the original locale
 			// in string-arrays, we fill the non-translated ones with original locale string.
 			$value = $entry->translations[0];
 
-			if ( ! $value )
+			// If we don't have a value for the translation, use the singular.
+			if ( ! $value ) {
 				$value = $entry->singular;
+			}
 
+			// Add the entry to the mapping array after escaping it.
 			$mapping[ $array_name ][] = $this->escape( $value );
 		}
 
-		foreach ( array_keys( $mapping ) as $array_name )
-		{
+		// Now do the actual output to the class variable.
+		foreach ( array_keys( $mapping ) as $array_name ) {
+			// Open the string array tag.
 			$this->line( '<string-array name="' . $array_name . '">', 1 );
 
-			foreach ( $mapping[ $array_name ] as $item )
-			{
-				$this->line('<item>' . $item . '</item>', 2);
+			// Output each item in the array.
+			foreach ( $mapping[ $array_name ] as $item ) {
+				$this->line( '<item>' . $item . '</item>', 2 );
 			}
 
+			// Close the string array tag.
 			$this->line( '</string-array>', 1 );
 		}
 	}
 
+	/**
+	 * Output the plural entries to the exported class variable.
+	 *
+	 * @since 2.4.0
+	 *
+	 * @param obj $entries The entries to store.
+	 * @param obj $locale  The locale object to use for exporting.
+	 */
+	private function plurals( $entries, $locale ) {
+		$nplurals = $locale->nplurals;
+		$mapping = array();
+		$order = $this->get_plural_order( $locale );
+
+		// Sort the entries before processing them.
+		uasort( $entries, array( $this, 'cmp_context' ) );
+
+		// Loop through all of the single entries add them to a mapping array.
+		foreach ( $entries as $entry ) {
+			// Make sure the array name is sanitized.
+			$array_name = preg_replace( '/\[\d+\]$/', '', $entry->context );
+
+			// Initialize the mapping array entry if this is the first time.
+			if ( ! isset( $mapping[ $array_name ] ) ) {
+				$mapping[ $array_name ] = array();
+			}
+
+			// Add the entry to the mapping array after escaping it.
+			$mapping[ $array_name ] = $entry;
+		}
+
+		// Now do the actual output to the class variable.
+		foreach ( $mapping as $array_name => $item ) {
+			// Open the string array tag.
+			$this->line( '<plurals name="' . $array_name . '">', 1 );
+
+			// Output each item in the array.
+			for ( $i = 0; $i < $nplurals; $i++ ) {
+				$this->line( '<item quantity="'. $order[ $i ] . '">' . $item->translations[ $i ] . '</item>', 2 );
+			}
+
+			// Close the string arrary tag.
+			$this->line( '</plurals>', 1 );
+		}
+	}
+
+	/**
+	 * Determine the order of plurals with relation to the Android standard.
+	 * See https://developer.android.com/guide/topics/resources/string-resource.html#Plurals for details.
+	 *
+	 * @since 2.4.0
+	 *
+	 * @param obj $locale  The locale object to use for exporting.
+	 */
+	private function get_plural_order( $locale ) {
```

Comment:
Yes, the code in that function is primarily a placeholder at the moment, it needs more work to handle the cases better. 

In the end we might have to add the Android types to the locales definitions (I'd like to avoid that if possible).